src/MongoDB.Bson/IO/JsonWriter.cs

```diff
@@ -648,13 +717,15 @@ public override void WriteTimestamp(long value)
                 ThrowInvalidState("WriteTimestamp", BsonWriterState.Value, BsonWriterState.Initial);
             }
 
-            var secondsSinceEpoch = (int)((value >> 32) & 0xffffffff);
-            var increment = (int)(value & 0xffffffff);
+            var secondsSinceEpoch = (uint)((value >> 32) & 0xffffffff);
```

Comment:
Note: this is the part of changes made for timestamp to be serializable in unsigned representation.